aioleakybucket/dynamic/__init__.py

```diff
@@ -0,0 +1,70 @@
+import asyncio
+from .zone import TooManyRequestsError, Zone
+
+
+class RequestLimiter:
+
+    def __init__(self, zones):
+        self.zones = {zone_name: Zone(**zones[zone_name]) for zone_name
+                      in zones}
+        self.resources = {}
+
+    def get_request_delay(self, requester_id, resource):
+        zone = self.zones[resource['zone']]
+        return zone.get_request_delay(requester_id, resource['burst'],
+                                      resource['delay'])
+
+    def get_resource(self, name, burst, delay)
+        if name not in self.resources:
+            self.resources[name] = {
+                'burst': burst,
+                'delay': delay,
+            }
+        return self.resources[name]
+
+
+limiter = None
+
+def setup(zones):
+    if limiter is None:
+        limiter = RequestLimiter(zones)
```

Comment:
Singletone limiter object for entire app. Yes, it can be initialized explicitly, but this function basically helps to avoid double initialization

Reply:
Well, this wouldn't work because `limiter` is a local variable here. You can force it to work with `global` keyword but why do you need that singleton?